tests/test_utils_life_rules.cairo

```diff
@@ -0,0 +1,506 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import (HashBuiltin,
+    BitwiseBuiltin)
+from starkware.cairo.common.alloc import alloc
+
+from contracts.utils.life_rules import get_adjecent, evaluate_rounds
+
+@external
+func test_get_adjecent{
+        range_check_ptr
+    }():
+    let (L, R, U, D, LU, RU, LD, RD) = get_adjecent(16)
+    assert L = 15
+    assert R = 17
+    assert U = 1
+    assert D = 31
+    assert LU = 0
+    assert RU = 2
+    assert LD = 30
+    assert RD = 32
+    return ()
+end
+
+@external
+func test_get_adjecent_wrapped{
+        range_check_ptr
+    }():
+    let (L, R, U, D, LU, RU, LD, RD) = get_adjecent(0)
+    assert L = 14
+    assert R = 1
+    assert U = 210
+    assert D = 15
+    assert LU = 224
+    assert RU = 211
+    assert LD = 29
+    assert RD = 16
+    return ()
+end
+
+@external
+func test_evaluate_rounds{
```

Comment:
I think we should check more edge (literally! 😆) cases

Reply:
Since evaluate rounds just calls apply_rules, and apply_rules just takes what get_adjecent returns, I tested edges in get_adjecent test cases, I guess it should be enough for now. I will try to add more tests later on though (perhaps aftrer indexer work)